script.js

```diff
@@ -1 +1,402 @@
-console.log('yay!');
+// Global setup ----------------------------------------------------
+let board = [];
+let firstCard = null;
+const boardSize = 4; // has to be an even number
+const matchesNeededToWin = (boardSize * boardSize) / 2;
+let numOfMatches = 0;
+let deck;
+let playerName;
+let secondsLeftTillGameOver = 180;
+let numOfGames = 1;
+const loseMessage = 'Sorry but the time ran out! Click on a square to try again!';
+let timerId; // to store ID of startMatchTimer
+
+let gameStatus = 'playGame';
```

Comment:
Constant statuses like 'playGame' should be stored in constant variables to avoid bugs due to typos. Consider defining constants like `const GAME_STATUS_PLAY_GAME = 'GAME_STATUS_PLAY_GAME'` above where we first declare and initialise `gameStatus`, and using the constants instead of inline strings any time we refer to the relevant game status.